temporalcli/client.go

```diff
@@ -157,6 +157,14 @@ func clientIdentity() string {
 
 type stringMapHeadersProvider map[string]string
 
+func newStringMapHeadersProvider() stringMapHeadersProvider {
+	rmap := make(map[string]string)
+	if authHeader := os.Getenv(`TEMPORAL_CLI_AUTHORIZATION_TOKEN`); authHeader != "" {
```

Comment:
aren't we using `TEMPORAL_X` for env vars instead of `TEMPORAL_CLI_X`? I already switched all my scripts and everything...

Reply:
Good question. There aren't any in the CLI rewrite so far, and the only one in the repo's main branch is https://github.com/temporalio/cli/blob/main/common/defs.go#L38

Which uses `CLI`. I assume the env vars you're talking about are for server itself? Seems reasonable to draw the distinction here, plus this is the same name used for this concept back in `tctl`, not that that necessarily matters much.

I'm fine either way. Will avoid merging this for a bit to see if anyone else has opinions.

Reply:
For tctl they were all `TEMPORAL_CLI_`, for temporal (old new cli) they were all changed to `TEMPORAL_`. It caused a small amount of hassle but now that we've done that everywhere, I'd much rather keep them `TEMPORAL_`

Reply:
There are a bunch in the main branch: https://github.com/search?q=repo%3Atemporalio%2Fcli+TEMPORAL_&type=code

Reply:
> There aren't any in the CLI rewrite so far

There are, but they are all tied to command options, so you'll find them in `commands.md`. They are just `TEMPORAL_X`. I would recommend `TEMPORAL_AUTH_HEADER` here.

I personally think a better approach to this whole thing is to do https://github.com/temporalio/cli/issues/478 and give that an env var for that option (ideally waiting on Go SDK to implement it first). Almost everyone will be doing `Bearer <token>` here anyways.

Reply:
> There aren't any in the CLI rewrite so far

There are, but they are all tied to command options, so you'll find them in `commands.md`. They are just `TEMPORAL_X`. I would recommend `TEMPORAL_AUTH_HEADER` here.

I personally think a better approach to this whole thing is to do https://github.com/temporalio/cli/issues/478 and give that an env var for that option (ideally waiting on Go SDK to implement it first). Almost everyone will be doing `Bearer <token>` here anyways.